_posts/2020-01-30-new-age-of-bastion.md

```diff
@@ -0,0 +1,165 @@
+
+I didn't aim for faster runtime, but I aimed to exploit the whole resources whenever it's possible. Bastion also doesn't aim for a being replacement to runtimes out there. It just unfolds a different approach to design systems. This system design is very graspable for the people who come from a functional background like Scala, Erlang, or Haskell. When we released the Bastion 0.1 in June 2019 with Tokio runtime, I realized that it's not the way that this project can continue. With premature design assumptions of the runtime, I struggled with developing the project with Tokio runtime. So we had a multiple run queue based statistics sampled async runtime. Throughout time, people wanted to use blocking tasks also. So we came with an idea of using an adaptive blocking pool, which can [automatically scale based on the throughput](https://docs.rs/bastion-executor/0.3.4/bastion_executor/blocking/index.html) and its' hogs up and down easily.
+
+## What is upcoming for Bastion?
+
+* We are currently working on being agnostic. Working in everywhere but still supplying our own executor to the people who want to have fault-tolerant systems in their projects.
+
+* We are exploring HCS(Hot code swap) case. And currently, we have [design discussions](https://github.com/bastion-rs/bastion/issues/103) going on.
+
+* WASM is a rising star in the ecosystem. We would like to support it, first we want to make everything `#[no_std]`, then we want to move towards to make Bastion projects compile for WASM.
+
+* We are working on Bastion Streams! A streaming framework implementation which is backpressure aware and works with push interface like API. Many people are asking about how this will be implemented, and I have pretty concrete ideas about this. Especially from the perspective of Rust! So now I am quite bit busy enabling parallel streams in Rust. **Not with [this](https://docs.rs/futures/0.3.1/futures/stream/trait.Stream.html), or [this](https://rust-lang.github.io/async-book/05_streams/01_chapter.html), or via [this](https://rust-lang.github.io/async-book/06_multiple_futures/03_select.html)**. Come join us if you find this interesting!
+
+* Distributed properties are still needs to be solved. We are slow at that side but wrote initial epidemic protocol implementation in our [Artillery](http://github.com/bastion-rs/artillery) repository.
+
+## New version is in here! Tell me more!
+
+We have added features to `0.3.4`, you can see it in our [CHANGELOG](https://github.com/bastion-rs/bastion/blob/master/CHANGELOG.md). But I will showcase a bunch of excellent features of the new Bastion version!
+
+**At Bastion itself, at the surface API level, here are our changes:**
+
+0- We added restart strategies, linear and exponential backoff between restarts. And maximum restarts for the failed children. Check out our [example in the repo](https://github.com/bastion-rs/bastion/blob/master/bastion/examples/restart_strategy.rs).
```

Comment:
Maybe we can thank the PR contributor if they allow us to (it was a really great PR :D )